connector/src/main/java/org/jboss/as/connector/subsystems/resourceadapters/ResourceAdapterParser.java

```diff
@@ -376,16 +376,20 @@ protected WorkManager parseWorkManager(final ModelNode operation, final XMLStrea
                 case START_ELEMENT: {
                     switch (WorkManager.Tag.forName(reader.getLocalName())) {
                         case SECURITY: {
-                            WM_SECURITY.parseAndSetParameter("true", operation, reader);
                             switch (Namespace.forUri(reader.getNamespaceURI())) {
                                 case RESOURCEADAPTERS_1_0:
                                 case RESOURCEADAPTERS_1_1:
                                 case RESOURCEADAPTERS_2_0:
                                 case RESOURCEADAPTERS_3_0:
                                 case RESOURCEADAPTERS_4_0:
+                                    WM_SECURITY.parseAndSetParameter("true", operation, reader);
                                     security = parseWorkManagerSecurity(operation, reader);
                                     break;
+                                case RESOURCEADAPTERS_7_0:
+                                    security = parseWorkManagerSecurity_7_0(operation, reader);
+                                    break;
                                 default: // If the switch statement contains a default it does not need to be revisited each time a new schema is added unless it actually affects it.
+                                    WM_SECURITY.parseAndSetParameter("true", operation, reader);
```

Comment:
This switch statement is back to front, the default needs to be the latest parser.

As this stands when version 8 of this schema is added it will revert to calling version 5 of this parse method.

Instead you need a case for 5 and 6 that calls version 5 of the parse method, then default then calls version 7 so it is used for 7 and later until the next time a change is made.

Also adding these parseAndSetParameter calls makes the switch statement more complication, I think those will be better at the top of the parse methods and then obviously not at the top of the method for 7.